### PR TITLE
USHIFT-3457: Various improvements of "Release notes" job

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

File: scripts/release-notes/gen_ec_release_notes.py [hunk 1]
```diff
@@ -356,10 +392,10 @@ def check_for_new_releases(url_base, release_type, version):
             break
     else:
         rpm_names = ',\n'.join(rpm_list)
-        print(f"WARNING: Did not find {microshift_rpm_name_prefix} in {rpm_names}")
+        logging.warning(f"WARNING: Did not find {microshift_rpm_name_prefix} in {rpm_names}")
```
Comment (pacevedom) [inline scripts/release-notes/gen_ec_release_notes.py:395]: Dont we get the log level automatically with `logging`? Do we need the `WARNING` in the message too?

File: scripts/release-notes/gen_ec_release_notes.py [hunk 2]
```diff
@@ -317,7 +352,7 @@ def find_new_releases(versions_to_scan, url_base, release_type):
             if nr:
                 new_releases.append(nr)
         except Exception as err:  # pylint: disable=broad-except
-            print(f"WARNING: could not process {release_type} {version}: {err}")
+            logging.error(f"WARNING: could not process {release_type} {version}: {err}")
```
Comment (pacevedom) [inline scripts/release-notes/gen_ec_release_notes.py:355]: Error or warning?

File: scripts/release-notes/gen_ec_release_notes.py (hunk 1 above)
Comment (pmtk) [reply scripts/release-notes/gen_ec_release_notes.py]: Missed that, fixed!

File: scripts/release-notes/gen_ec_release_notes.py (hunk 2 above)
Comment (pmtk) [reply scripts/release-notes/gen_ec_release_notes.py]: Changed to warning and removed `WARNING: ` from msg

**Review by pacevedom — APPROVED** [review]
/lgtm

**Pull request merged**